src/uct/ib/base/ib_iface.c

```diff
@@ -494,3 +502,63 @@ struct ibv_ah *uct_ib_create_ah(uct_ib_iface_t *iface, uint16_t dlid)
 
     return ibv_create_ah(dev->pd, &ah_attr);
 }
+
+int uct_ib_iface_open_fd(uct_iface_t *tl_iface)
+{
+    uct_ib_iface_t *iface = ucs_derived_of(tl_iface, uct_ib_iface_t);
+    uct_ib_device_t *dev = uct_ib_iface_device(iface);
+
+    uct_ib_iface_comp_t *comp = ucs_malloc(sizeof(*comp),
+            "completion channel");
+    if (comp == NULL) {
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    comp->comp_channel = ibv_create_comp_channel(dev->ibv_context);
```

Comment:
need to pass comp_channel to ibv_create_cq, and also use ibv_req_notify_cq to arm the event